code/modules/client/mouse_pointer/_mouse_pointer.dm

```diff
@@ -0,0 +1,69 @@
+/*
+ * Adds pointer entries to clients to allow for multiple sources wanting to modify the pointer at once.
+ * - add_mouse_pointer(pointer_type, pointer_priority, icon_index) will add or replace a pointer of the specified /decl/mouse_pointer type.
+ * - remove_mouse_pointer(pointer_type) will clear that entry.
+ * - Updates are handled automatically by adding/removing, other procs should not be used without a good reason.
+ */
+
+/client
+	VAR_PRIVATE/list/_mouse_pointers
+
+/client/proc/clear_mouse_pointers()
+	if(LAZYLEN(_mouse_pointers))
+		LAZYCLEARLIST(_mouse_pointers)
+		update_mouse_pointer()
+		return TRUE
+	return FALSE
+
+/client/proc/set_mouse_pointer_icon(new_pointer)
+	if(isnull(new_pointer))
+		new_pointer = initial(mouse_pointer_icon)
+	if(mouse_pointer_icon != new_pointer)
+		mouse_pointer_icon = new_pointer
+		return TRUE
+	return FALSE
+
+/client/proc/add_mouse_pointer(pointer_type, pointer_priority = 1, icon_index = 1)
+
+	var/decl/mouse_pointer/pointer_decl = GET_DECL(pointer_type)
+	if(!isnum(icon_index) || icon_index < 1 || icon_index > length(pointer_decl.icons))
+		CRASH("Invalid icon_index passed to add_mouse_pointer() for [pointer_type].")
+
+	// Is an identical pointer already being tracked?
+	var/pointer_icon = pointer_decl.icons[icon_index]
+	if(pointer_type in _mouse_pointers)
+		var/list/comparing = _mouse_pointers[pointer_type]
+		if(comparing["icon"] == pointer_icon && comparing["priority"] == pointer_priority)
+			return FALSE
+
+	// Update our list entry. If we have multiple pointers, sort by priority.
+	var/need_update = !(pointer_type in _mouse_pointers)
+	LAZYINITLIST(_mouse_pointers)
+	_mouse_pointers[pointer_type] = list("icon" = pointer_icon, "priority" = pointer_priority)
+	if(LAZYLEN(_mouse_pointers) > 1)
+		_mouse_pointers = sortTim(_mouse_pointers, /proc/cmp_priority_list, TRUE)
+		need_update = TRUE
+
+	// Refresh if needed.
+	if(need_update)
+		update_mouse_pointer()
+
+/client/proc/remove_mouse_pointer(pointer_type)
+	if(!LAZYLEN(_mouse_pointers) || !_mouse_pointers[pointer_type])
```

Comment:
could be written more succinctly by using the null indexing operator. just down to personal preference though
```suggestion
	if(!_mouse_pointers?[pointer_type])
```